bench/BenchAll.hs

```diff
@@ -11,15 +11,23 @@
 {-# LANGUAGE PackageImports      #-}
 {-# LANGUAGE ScopedTypeVariables #-}
 {-# LANGUAGE MagicHash           #-}
+{-# LANGUAGE ViewPatterns        #-}
 
 module Main (main) where
 
+import           Control.Exception                     (assert)
 import           Data.Foldable                         (foldMap)
 import           Data.Monoid
 import           Data.Semigroup
 import           Data.String
 import           Test.Tasty.Bench
 import           Prelude                               hiding (words)
+import           Numeric.IEEE
+import           GHC.Float                             (powerFloat,
+                                                        castWord32ToFloat,
+                                                        castWord64ToDouble,
+                                                        castFloatToWord32,
+                                                        castDoubleToWord64)
```

Comment:
These functions are buggy with old versions of ghc. You want the versions in `Data.ByteString.Builder.Prim.Internal.Floating`. But I guess that requires a bit of surgery on the package description to make available here... perhaps just vendor trivial `\x -> unsafePerformIO $ with x (peek . castPtr)` implementations like the testsuite does.